x/ibc/23-commitment/codec.go

```diff
@@ -0,0 +1,18 @@
+package commitment
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+// RegisterCodec registers types declared in this package
+func RegisterCodec(cdc *codec.Codec) {
+	cdc.RegisterInterface((*RootI)(nil), nil)
+	cdc.RegisterInterface((*PrefixI)(nil), nil)
+	cdc.RegisterInterface((*PathI)(nil), nil)
+	cdc.RegisterInterface((*ProofI)(nil), nil)
+
+	cdc.RegisterConcrete(Root{}, "ibc/commitment/merkle/Root", nil)
```

Comment:
ICS23 is supposed to be a generic format, not just for the sdk, but also for agoric (js), for rust chains (polkadot), etc.

As there is no truly cross-language support for go-amino, I wonder about the usage here. Also, the very unclear definition of the pieces (`merkle.Proof` used below is a (string, []byte) blob without clear spec.

I can see this approach for a quick prototype, but if you are attempting a 1.0 of something interoperable, there needs to be a clear spec and possibility to implement in other languages. I spent considerable time getting that to work in github.com/confio/ics23 and I wonder why that was not used (not comments left on that repo to explain what was lacking/wrong with that approach)?

Reply:
@mossid @fedekunze ^^

Reply:
@ethanfrey correct, afaik for the IBC completeness your code should be the one that we’ll need to integrate here. With respect to what was lacking I think @cwgoes created an issue somewhere (not sure if it was here or on the ICS repo)

Reply:
Yes, that's right - the plan is to integrate confio/ics23 prior to 1.0.0. - https://github.com/cosmos/cosmos-sdk/issues/5199